.circleci/generate.sh

```diff
@@ -145,52 +141,38 @@ if $has_env_vars && $check_env_vars; then
   done
 fi
 
-if $lowres; then
-  ($all || $midres || $highres) && die "Cannot use option -l with options -a, -m or -h"
-  echo "Generating new config.yml file with low resources from config-2_1.yml"
-  circleci config process $BASEDIR/config-2_1.yml > $BASEDIR/config.yml.LOWRES.tmp
-  cat $BASEDIR/license.yml $BASEDIR/config.yml.LOWRES.tmp > $BASEDIR/config.yml
-  rm $BASEDIR/config.yml.LOWRES.tmp
-
-elif $midres; then
-  ($all || $lowres || $highres) && die "Cannot use option -m with options -a, -l or -h"
-  echo "Generating new config.yml file with middle resources from config-2_1.yml"
-  patch -o $BASEDIR/config-2_1.yml.MIDRES $BASEDIR/config-2_1.yml $BASEDIR/config-2_1.yml.mid_res.patch
-  circleci config process $BASEDIR/config-2_1.yml.MIDRES > $BASEDIR/config.yml.MIDRES.tmp
-  cat $BASEDIR/license.yml $BASEDIR/config.yml.MIDRES.tmp > $BASEDIR/config.yml
-  rm $BASEDIR/config-2_1.yml.MIDRES $BASEDIR/config.yml.MIDRES.tmp
+if $free; then
+  ($all || $paid) && die "Cannot use option -f with options -a or -p"
+  echo "Generating new config.yml file for free tier from config-2_1.yml"
```

Comment:
For my own learning: why is this file named `config-2_1.yml`. 2.1 of what?

Reply:
I don't know for sure, but I think it refers to the version of CircleCI's config that it contains, which is 2.1 ([first line of the file](https://github.com/apache/cassandra/blob/trunk/.circleci/config-2_1.yml#L19)). 

The `config.yml` file contains the expanded config that is automatically generated by `circle config process`. That file however [uses version 2.0](https://github.com/apache/cassandra/blob/trunk/.circleci/config.yml#L19) instead of 2.1, don't ask me why. It has to be named just `config.yml` so Circle is able to identify it.